src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -536,6 +624,74 @@ public void sendActionListenerOnFailureRequest(TransportService transportService
+     *
+     * @param componentSettings The component setting that correspond to the values provided by the environment settings
+     * @param transportService  The TransportService defining the connection to OpenSearch.
+     */
+    public void sendEnvironmentSettingsRequest(TransportService transportService, List<Setting<?>> componentSettings) {
+        logger.info("Sending Environment Settings request to OpenSearch");
+        EnvironmentSettingsResponseHandler environmentSettingsResponseHandler = new EnvironmentSettingsResponseHandler();
+        try {
+            transportService.sendRequest(
+                opensearchNode,
+                ExtensionsOrchestrator.REQUEST_EXTENSION_ENVIRONMENT_SETTINGS,
+                new EnvironmentSettingsRequest(componentSettings),
+                environmentSettingsResponseHandler
+            );
+        } catch (Exception e) {
+            logger.info("Failed to send Environment Settings request to OpenSearch", e);
+        }
+    }
+
+    /**
+     * Registers the component {@link Setting} and the corresponding consumer to the settingsUpdateConsumer map.
+     * Extensions will use this method instead of invoking ClusterSettings#addSettingsUpdateConsumer
+     *
+     * @param componentSetting The component setting associated with the consumer
+     * @param consumer  The setting update consumer associated with the component setting
+     */
+    public <T> void registerSettingUpdateConsumer(Setting<T> componentSetting, Consumer<T> consumer) {
+        this.settingUpdateConsumers.put(componentSetting, consumer);
+    }
+
+    /**
+     * Requests the environment setting values from OpenSearch for the corresponding component settings.
+     * This should be called after extension components have registered all their setting update consumers with the settingUpdateConsumers map.
+     * The result will be handled by a {@link EnvironmentSettingsResponseHandler}.
+     *
+     * @param transportService  The TransportService defining the connection to OpenSearch.
+     * @throws Exception if there are no setting update consumers registered within the settingUpdateConsumers map
+     */
+    public void sendAddSettingsUpdateConsumerRequest(TransportService transportService) throws Exception {
+        logger.info("Sending Add Settings Update Consumer request to OpenSearch");
+
+        // Determine if setting update consumers have been registered
+        if (this.settingUpdateConsumers.isEmpty()) {
+            throw new Exception("There are no setting update consumers registered");
+        } else {
+
+            // Extract registered settings from setting update consumer map
+            List<Setting<?>> componentSettings = new ArrayList<>(this.settingUpdateConsumers.size());
+            componentSettings.addAll(this.settingUpdateConsumers.keySet());
+
+            AddSettingsUpdateConsumerResponseHandler addSettingsUpdateConsumerResponseHandler =
```

Comment:
Can we use `ExtensionBoolerHandler` here?

Reply:
Sure, I'll replace this and delete the AddSettingsUpdateConsumerResponseHandler